app/controllers/to_do_lists_controller.rb

```diff
@@ -1,10 +1,11 @@
 class ToDoListsController < ApplicationController
 
   def index
-    lists = Finder.new(cookies[:filter]).perform
-    @lists_presentation = ListsPresentation.new lists
+    @lists_per_page = Finder.new(cookies[:filter]).perform
+    @lists_per_page = @lists_per_page.paginate(page: params[:page])
+    @lists_presentation = ListsPresentation.new @lists_per_page
 
-    @lists = lists.map { |list| SingleListPresenter.new list }
+    @lists = @lists_per_page.map { |list| SingleListPresenter.new list }
```

Comment:
@sergeyradkevich, unfortunately can't, because `@lists_per_page` has a lot of additional methods thanks to pagination, like `current_page`, `total_pages`,`per_page`, `limit`, `offset`, etc. So it is not possible to paginate through the array.